data_structures/trie/radix_tree.py

```diff
@@ -57,7 +57,7 @@ def insert(self, word: str) -> None:
         """
         # Case 1: If the word is the prefix of the node
         # Solution: We set the current node as leaf
-        if self.prefix == word:
+        if self.prefix == word and not self.is_leaf:
```

Comment:
Please add a doctest that proves the old approach was broken and the new approach works as expected.